TwoPageEBookReader/app/src/main/java/com/example/ebook_reader_sample/Book.kt

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ */
+package com.example.ebook_reader_sample
+
+import android.content.Context
+import android.content.res.AssetManager
+import android.graphics.Rect
+import android.text.Html
+import android.view.View
+import android.widget.LinearLayout
+import android.widget.TextView
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import kotlin.math.max
+import kotlin.math.min
+
+class Book(private val filePath: String, private val activityContext: Context) {
+    private val assetManager: AssetManager = activityContext.assets
+    private var chapterStarts: ArrayList<Int> = ArrayList()
+    private var chapterLengths: ArrayList<Int> = ArrayList()
+    val numChapters
+        get() = chapterStarts.size
+    var chapterStartString = activityContext.resources.getString(R.string.chapter_start)
+    var chapterEndString = activityContext.resources.getString(R.string.chapter_end)
+
+    init {
+        val bufferedReader = BufferedReader(InputStreamReader(assetManager.open(filePath)))
+
+        var line = 0
+        var lineCount = 0
+        var lineString: String? = ""
+        var emptyLineCount = 2
+
+        do {
+            if (lineString!!.isEmpty()) {
+                emptyLineCount += 1
+            } else {
+                if (emptyLineCount >= 3) {
+                    chapterStarts.add(line)
+                    chapterLengths.add(lineCount)
+                    lineCount = 0
+                }
+                emptyLineCount = 0
+            }
+            line += 1
+            lineCount += 1
+            lineString = bufferedReader.readLine()
+        } while (lineString != null)
+        bufferedReader.close()
+
+        chapterLengths.add(lineCount)
+        chapterLengths.removeAt(0)
+    }
+
+    private lateinit var paragraphStrings: ArrayList<String>
+    val chapterTitle
+        get() = "(Section ${currentChapter}) ${paragraphStrings[0]}"
+
+    var currentChapter = 0
+        set(inChapter) {
```

Comment:
Same here, this is too long and complex for a set variable, is better if you move the code to its own function so you have just:
```
var currentChapter =0 
   set(inChapter) {
      doSomething()
    }   
```

Reply:
moved to _setCurrentChapter() function